httpconnect/httpconnect.go

```diff
@@ -38,7 +38,24 @@ func New(opts *Options) filters.Filter {
 		}
 	}
 
-	return &httpConnectHandler{opts}
+	f := &httpConnectHandler{Options: opts}
+	f.ic = interceptor.New(&interceptor.Opts{
+		Dial:      f.dial,
+		GetBuffer: buffers.Get,
```

Comment:
I'm curious, why these buffer operations that are such an implementation detail provided as an option. Do you actually provide different "buffer getter/setter" somewhere?